Core/PARStore.m

```diff
@@ -1407,15 +1443,36 @@ - (BOOL)deleteBlobAtPath:(NSString *)path error:(NSError **)error
     // otherwise blobs are stored in a special blob directory
     __block NSError *localError = nil;
     NSURL *fileURL = [[self blobDirectoryURL] URLByAppendingPathComponent:path];
+    NSURL *tombstoneURL = [fileURL URLByAppendingPathExtension:TombstoneFileExtension];
+
     NSError *coordinatorError = nil;
-    [[self newFileCoordinator] coordinateWritingItemAtURL:fileURL options:NSFileCoordinatorWritingForReplacing error:&coordinatorError byAccessor:^(NSURL *newURL)
+    
+    [[self newFileCoordinator] coordinateWritingItemAtURL:fileURL options:NSFileCoordinatorWritingForReplacing writingItemAtURL:tombstoneURL options:NSFileCoordinatorWritingForReplacing error:&coordinatorError byAccessor:^(NSURL * _Nonnull newURL, NSURL * _Nonnull newTombstoneURL)
      {
-         // write to disk (overwrite any file that was at that same path before)
-         NSError *error = nil;
-         BOOL success = [[NSFileManager defaultManager] removeItemAtURL:fileURL error:&error];
-         if (!success)
-             localError = [NSError errorWithObject:self code:__LINE__ localizedDescription:[NSString stringWithFormat:@"Could not delete data blob at path '%@'", newURL.path] underlyingError:error];
-     }];
+        // TODO: Decide how to interpret a tombstone file being present already.
+        // With the previous implementation, I think that calling delete on a file that didn't exist
+        // (or had already been deleted) returned an error.
+        // To be consistent with this behaviour, we should perhaps also return an error if we
+        // find an existing tombstone file.
+        // However, my normal inclination would be to treat the presence of the tombstone as a sign
+        // that the file has been successfully deleted and quietly return success.
+        
+        NSError *error = nil;
+
+        BOOL success = !usingTombstone;
+        if (!success) {
+            // write tombstone
+            success = [self writeTombstoneAtPath:newTombstoneURL.path forFileAtPath:newURL.path error:&error];
+            if (!success)
+                localError = [NSError errorWithObject:self code:__LINE__ localizedDescription:[NSString stringWithFormat:@"Could not create tombstone for data blob at path '%@'", newURL.path] underlyingError:error];
```

Comment:
Think I would prefer having the {} on the if, although it is long ago that I wrote ObjC.

Reply:
I agree but I was trying to match the style of the other code. That line was copied & pasted from elsewhere in fact (then the error was changed).

Reply:
OK, no problem. If that is the style of the rest, happy to keep it that way.